### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by anyx**
- Removed `file` method
- `$options` argument for every request method
- `$extraUrl` and `$parameters` moved into `$options`
- requirement php 7.1 and symfony 4.3

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,7 +7,11 @@
     "authors": [
         {"name": "Vyacheslav Salakhutdinov", "email": "megazoll@gmail.com"}
     ],
-    "require": {},
+    "require": {
+        "php": "^7.1.3",
```
Comment (megazoll) [inline composer.json]: `>=7.1`

File: composer.json [hunk 2]
```diff
@@ -7,7 +7,11 @@
...
+        "ext-json": "*",
+        "symfony/framework-bundle": "4.3.*"
```
Comment (megazoll) [inline composer.json]: `^4.3`

File: RequestResult.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 namespace Staffim\RestClient;
 
-class Response
+class RequestResult
```
Comment (megazoll) [inline RequestResult.php]: maybe rename `getResponse` in `getRaw` or `getRawResponse`?

Comment (anyx) [reply RequestResult.php]: updated

File: composer.json (hunk 2 above)
Comment (anyx) [reply composer.json]: fixed

**Pull request merged**